### DISCUSSION TITLE
`auto_show = false` only for cmdline?

### DISCUSSION THREAD

**Discussion by Julian** [saghen/blink.cmp#681] [Q&A] [answered]
Is there a way to disable `auto_show` only for cmdline completion?

In particular it's quite annoying to me to not be able to use history navigation (up / down arrows) without dismissing the completion -- the behavior I want is that up/down do what they normally do in command mode (scroll backwards through history), and that that change only if I hit tab to complete something.

Anyone know if that's configurable at the minute?

**Comment by quebn** [comment]
i want something similar too, in my case i want the cmdline completion to not show unless i press tab.

**Comment by edr3x** [comment]
@Julian if you want to disable the cmdline itself then 
```lua
    opts = {
        sources = {
            cmdline = {
                enabled = false,
            },
        },
    }
```

**Reply by Julian** [reply]
I don't want to disable it entirely ideally. Though I'm finding quite a few other issues with it so I'm close to reconsidering (which isn't surprising, I found nvim-cmp's command line completion also not very usable). But the original point of the question was to keep it on. Thanks for the info though.

**Comment by richin13** [comment, marked as answer]
I found this recipe in the docs: [Don't show completion menu automatically in cmdline mode ](https://cmp.saghen.dev/recipes.html#don-t-show-completion-menu-automatically-in-cmdline-mode)

```lua
completion = { 
  menu = { auto_show = function(ctx) return ctx.mode ~= 'cmdline' end }
}
```

**Reply by Julian** [reply]
Nice! That definitely looks relevant. I've since disabled command line completion but will give it a shot, thanks for the pointer.
